crates/partition-store/src/outbox_table/mod.rs

```diff
@@ -82,15 +82,16 @@ fn get_outbox_message<S: StorageAccess>(
 fn truncate_outbox<S: StorageAccess>(
     storage: &mut S,
     partition_id: PartitionId,
-    seq_to_truncate: Range<u64>,
+    range: RangeInclusive<u64>,
 ) {
-    let mut key = OutboxKey::default().partition_id(partition_id);
-    let k = &mut key;
-
-    for seq in seq_to_truncate {
-        k.message_index = Some(seq);
-        storage.delete_key(k);
-    }
+    let _x = RocksDbPerfGuard::new("truncate-outbox");
+    range
+        .map(|sn| {
+            OutboxKey::default()
+                .partition_id(partition_id)
+                .message_index(sn)
+        })
+        .for_each(|key| storage.delete_key(&key));
```

Comment:
Out of curiosity: Why did you change the explicit for loop into a map-for_each?

Reply:
I wrote the loop from scratch for the second PR revision, but I see now that the original was much simpler! I've moved it back to the for-loop form with the next iteration. I am just enjoying having sane iterator APIs after spending too long in JS land ;-)